### PR TITLE
Enable http status code propagation

### PR DESCRIPTION
**Pull request by smcquay**
Here we consolidate json encoding in a slightly different way, one that allows us to propagate http status code in our success and error responses from handler functions.

I'm not super convinced of the package name(`jh`, I guess short for "json http"?), and would eventually move it out into its own repository if we find it useful. I'm looking for feedback on the approach at large, specifics about this implementation, and how the repository has changed to react to doing it according to this new paradigm.

### CODE REVIEW

File: cmd/emctl/inject.go [hunk 1]
```diff
@@ -37,7 +50,8 @@ func inject(ctx context.Context, c pb.EventMasterClient) error {
 			if err != nil {
 				return errors.Wrap(err, "add event")
 			}
-			log.Printf("%+v", resp.Id)
+			log.Printf("%v %v %v", resp.ID, dc, topic)
+			time.Sleep(2 * time.Second)
```
Comment (jacksontj) [inline cmd/emctl/inject.go]: Why the sleep?

File: jh/adapter.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+		if err := json.NewEncoder(w).Encode(&d); err != nil {
+			log.Printf("%v: json encode failure: %+v", req.URL.Path, err)
+		}
+	}
+}
+
+// RouterToStd exists to convert nominal http.HandlerFunc to httprouter.Handle.
+func RouterToStd(f httprouter.Handle) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
```
Comment (jacksontj) [inline jh/adapter.go]: can we just use https://godoc.org/github.com/julienschmidt/httprouter#Router.ServeHTTP ?

File: cmd/emctl/inject.go (hunk 1 above)
Comment (smcquay) [reply cmd/emctl/inject.go]: If the changes are bunched up too closely together it doesn't render nicely at "reasonable resolutions" in Grafana. It works as expected in `eventmaster`, but this is just a utility subcommand to inject some events to be rendered in Grafana, so I added the sleep.

File: jh/adapter.go (hunk 2 above)
Comment (smcquay) [reply jh/adapter.go]: Good point; I'll update.

**Pull request merged**